### PR TITLE
Create Sections for Journalist, Source, and Admin Guides

### PR DESCRIPTION
**Pull request by nathandyer**
## Status
Ready for review

## Description of Changes

This branch lays the foundation for an expansion of our Source and Journalist Guides. This PR does **not** provide new content under either of those guides, but instead breaks apart existing material into separate, standalone articles under newly created sections. The benefit of this change is that it:
* makes the document submission flow clearer to Sources
* better emphasizes the necessary preparation steps and security concerns to submit documents as safely as possible
* makes the Journalist user section much easier to read
* clarifies that the document handing process is separate from the retrieval and decryption steps, and can be used as an independent resource for that specific topic
* makes future documentation expansion plans possible

With the ever-expanding table-of-contents, this branch also includes a small tweak to the index/welcome page that refers users directly to the relevant places in the documentation, and hides the duplicate table of contents. This does not represent the final vision for the index page, but rather servers as a necessary placeholder. If requested, that change can certainly be split out from this PR.

Given the expansive nature of this PR, I would prefer to hold off merging until at least @eloquence and @zenmonkeykstop both give approval, with the note that additional reviews and comments/feedback from others are encouraged and welcome.

## Testing
* [ ] CI Passes
* [ ] Builds locally

## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [x] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

### CODE REVIEW

**Comment by eloquence** [comment]
Thanks for your work on this, I definitely think this pushes us in the right direction. A couple of notes:

- I suggest we keep track of any 404s produced by the new structure. We should be able to set up redirects for those, so that any external links won't need to be updated.

- I think the "SecureDrop for Journalists" page will need to be a bit expanded in this new structure because it is a very inviting entrypoint, but in its current form it is only useful for journalists who've already set up the system. Of course this risks duplicating content from the general overview section, but perhaps we can at least add a short paragraph explaining the benefits of the system for journalists, and linking elsewhere as needed.

- What stands out to me as potentially confusing/disorienting in the remaining structure is the "Admin Guide" standalone section followed by Installation/Deployment/Maintenance. Some ideas around this:

 - Structurally, I think we still have an inherent issue with the heavy representation of admin content. Some ideas around that:

   - What do you think of moving the "Admin Guide" section after "Installation" and unpacking it so the top-level headings are visible in the table of contents? The thinking is that you first have to install the system before you can administer it, and this would make the "Admin Guide" section less of an odd duck in the table of contents.
  
   - Alternatively, I'm wondering if, with the tooling we have or small changes to it, we could introduce an additional level of nesting in the hierarchy, so that the main table of contents shows only a single "Admin Guide" section, which has collapsed detail sections for installation, administration, maintenance, etc. I'm aware that RTD seems to intentionally force a two level structure -- but since we're self-hosting, perhaps we can modify it to make it more suitable for our needs?

File: docs/index.rst
Comment (zenmonkeykstop) [inline docs/index.rst:13]: Duplicate paragraph?

Comment (nathandyer) [reply docs/index.rst]: Whoops! Fixed :)

**Comment by nathandyer** [comment]
Thank you @eloquence, I appreciated your notes! I definitely agree about keeping track of 404s. This PR makes a lot of big changes, and while most links should stay in-tact, it will be necessary to redirect where appropriate.

Like you, I was a bit bothered by how much of an odd duck the admin section was. At first, I tried to figure out how to make the hierarchy work with our tooling, but after some research and failed attempts, I realized I was trying to make Sphinx behave in a way it just wasn't designed to behave.

But then that made me re-think the structure, and I realized I hadn't gone far enough in properly organizing the admin section.

The two most recent commits are rather large, but they get the admin section in line with the new changes to the Source and Journalist sections.

I've restructured the files so that there is now an "admin" folder to match "source" and "journalist," and within the admin folder there are now subsections for "deployment," "installation," "maintenance," and "reference." I've updated the table of content headings to clarify which components are part of the admin guide, and to improve how it flows.

I also broke apart the first page of the admin section, which I realized was way too large. There are now separate documents for its component parts, which already makes a big difference in both structure and layout. These now changes also better align our documentation with the Divio system.

Please take another look at a local build, and let me know if you think I'm on the right track. Thanks!

File: docs/admin/reference/responsibilities.rst
Comment (eloquence) [inline docs/admin/reference/responsibilities.rst:14]: In this new structure, I would suggest we add a bullet point "install SecureDrop" linking to the installation guide.

File: docs/admin/reference/admin_interface.rst
Comment (eloquence) [inline docs/admin/reference/admin_interface.rst:73]: I think the FreeOTP and YubiKey headlines should be a level below "Adding users".

File: docs/admin/reference/ssh_access.rst
Comment (eloquence) [inline docs/admin/reference/ssh_access.rst:7]: To avoid a duplicate subtitle and make the section more introductory, I would change this subtitle to "Logging in via SSH" or similar, and add a brief reminder that they can log in via the commands `ssh app` or `ssh mon` from the Admin Workstation (future improvement opportunity: troubleshooting SSH access, but we don't have a guide for that yet).

**Comment by eloquence** [comment]
This is looking very clear, and I like the structure you've arrived at. Left some small nits, but otherwise looks great to me on a first pass.

Not strictly related to this PR, but a couple of other observations as I'm looking at the whole thing:
- For symmetry with the new docs structure, I would suggest introducing the user roles (section "User Roles") in the order in which they appear later: source, journalist, admin
- For the installation and deployment overview section, right now the installation overview is called "Installation Overview" and the deployment overview just "Overview". I would suggest either going with "Installation Overview / Deployment Overview", or "Overview / Overview".

Thanks much for all your hard work on this!

**Comment by nathandyer** [comment]
Thank you @eloquence! I believe I was able to address all your suggestions, but please let me know if I overlooked any of them.

File: docs/admin/reference/ssh_access.rst
Comment (eloquence) [inline docs/admin/reference/ssh_access.rst:8]: Nit: `ssh app` / `ssh mon` (currently rendered as italics due to single tic)

Structurally, I would put the instructions after the caveat ("Generally, you should avoid") and do it as regular text rather than in note format, for two reasons:
- as a general principle, I think it makes sense to show warnings/caveats before specific instructions (i.e. I would prefer "Warning: Pushing the button makes things explode. Step 1: Push the button" over "Step 1: Push the button. Warning: Pushing the button makes things explode" :)
- in this instance, we also have two "tip" call-outs competing for visual attention, so there's a risk of things getting a bit cluttered.

**Comment by eloquence** [comment]
Thanks Nathan! One nit, but otherwise this looks great. I really like the overall architecture you've arrived at here - it makes sense <3

File: docs/admin/reference/ssh_access.rst
Comment (nathandyer) [reply docs/admin/reference/ssh_access.rst]: Great catch! And I agree, that explanation makes sense, and after making the change I prefer it myself.

**Comment by nathandyer** [comment]
Thank you so much @eloquence! I'm glad these changes meet your approval! After having spent the week using this version of the docs, I still find myself enjoying these changes, which is usually a good sign. Assuming this gets merged, I hope our users feel the same!

Assuming the last commit addresses that last nit adequately, I'll leave everything for you to decide how to move forward. If you're comfortable giving an approval and merging straight away, that works great for me. But given the scope of this PR, if you want to hold off on merging for another review/approval, that makes sense too. Thanks for reviewing this, I know it was a big diff!

**Review by eloquence — APPROVED** [review]
LGTM! Given the scope of the changes, I think an additional review would be appropriate prior to merge.

If you can prepare a list of 404s that will result from the structural changes, I can prepare the redirect setup.

**Comment by nathandyer** [comment]
Thank you @eloquence!

I've put together this list, which I think covers all the URL changes:

Original URL | New URL
-- | --
https://docs.securedrop.org/en/stable/installation_overview.html | https://docs.securedrop.org/en/stable/admin/installation/installation_overview.html
https://docs.securedrop.org/en/stable/passphrases.html | https://docs.securedrop.org/en/stable/admin/installation/passphrases.html
https://docs.securedrop.org/en/stable/hardware.html | https://docs.securedrop.org/en/stable/admin/installation/hardware.html
https://docs.securedrop.org/en/stable/before_you_begin.html | https://docs.securedrop.org/en/stable/admin/installation/before_you_begin.html
https://docs.securedrop.org/en/stable/create_usb_boot_drives.html | https://docs.securedrop.org/en/stable/admin/installation/create_usb_boot_drives.html
https://docs.securedrop.org/en/stable/set_up_svs.html | https://docs.securedrop.org/en/stable/admin/installation/set_up_svs.html
https://docs.securedrop.org/en/stable/set_up_transfer_and_export_device.html | https://docs.securedrop.org/en/stable/admin/installation/set_up_transfer_and_export_device.html
https://docs.securedrop.org/en/stable/generate_submission_key.html | https://docs.securedrop.org/en/stable/admin/installation/generate_submission_key.html
https://docs.securedrop.org/en/stable/set_up_admin_tails.html | https://docs.securedrop.org/en/stable/admin/installation/set_up_admin_tails.html
https://docs.securedrop.org/en/stable/network_firewall.html | https://docs.securedrop.org/en/stable/admin/installation/network_firewall.html
https://docs.securedrop.org/en/stable/firewall_pfsense.html | https://docs.securedrop.org/en/stable/admin/installation/firewall_pfsense.html
https://docs.securedrop.org/en/stable/firewall_opnsense.html | https://docs.securedrop.org/en/stable/admin/installation/firewall_opnsense.html
https://docs.securedrop.org/en/stable/servers.html | https://docs.securedrop.org/en/stable/admin/installation/servers.html
https://docs.securedrop.org/en/stable/install.html | https://docs.securedrop.org/en/stable/admin/installation/install.html
https://docs.securedrop.org/en/stable/configure_admin_workstation_post_install.html | https://docs.securedrop.org/en/stable/admin/installation/configure_admin_workstation_post_install.html
https://docs.securedrop.org/en/stable/create_admin_account.html | https://docs.securedrop.org/en/stable/admin/installation/create_admin_account.html
https://docs.securedrop.org/en/stable/test_the_installation.html | https://docs.securedrop.org/en/stable/admin/installation/test_the_installation.html
https://docs.securedrop.org/en/stable/deployment_practices.html | https://docs.securedrop.org/en/stable/admin/deployment/deployment_practices.html
https://docs.securedrop.org/en/stable/deployment/landing_page.html | https://docs.securedrop.org/en/stable/admin/deployment/deployment/landing_page.html
https://docs.securedrop.org/en/stable/deployment/whole_site_changes.html | https://docs.securedrop.org/en/stable/admin/deployment/deployment/whole_site_changes.html
https://docs.securedrop.org/en/stable/deployment/sample_privacy_policy.html# | https://docs.securedrop.org/en/stable/admin/deployment/deployment/sample_privacy_policy.html
https://docs.securedrop.org/en/stable/getting_the_most_out_of_securedrop.html | https://docs.securedrop.org/en/stable/admin/deployment/getting_the_most_out_of_securedrop.html
https://docs.securedrop.org/en/stable/onboarding_journalists.html | https://docs.securedrop.org/en/stable/admin/deployment/onboarding_journalists.html
https://docs.securedrop.org/en/stable/onboarding_admins.html | https://docs.securedrop.org/en/stable/admin/deployment/onboarding_admins.html
https://docs.securedrop.org/en/stable/yubikey_setup.html | https://docs.securedrop.org/en/stable/admin/deployment/yubikey_setup.html
https://docs.securedrop.org/en/stable/https_source_interface.html | https://docs.securedrop.org/en/stable/admin/deployment/https_source_interface.html
https://docs.securedrop.org/en/stable/ssh_over_local_net.html | https://docs.securedrop.org/en/stable/admin/deployment/ssh_over_local_net.html
https://docs.securedrop.org/en/stable/remote.html | https://docs.securedrop.org/en/stable/admin/deployment/remote.html
https://docs.securedrop.org/en/stable/tails_printing_guide.html | https://docs.securedrop.org/en/stable/admin/deployment/tails_printing_guide.html
https://docs.securedrop.org/en/stable/offboarding.html | https://docs.securedrop.org/en/stable/admin/deployment/offboarding.html
https://docs.securedrop.org/en/stable/logging.html | https://docs.securedrop.org/en/stable/admin/maintenance/logging.html
https://docs.securedrop.org/en/stable/ossec_alerts.html | https://docs.securedrop.org/en/stable/admin/maintenance/ossec_alerts.html
https://docs.securedrop.org/en/stable/backup_and_restore.html | https://docs.securedrop.org/en/stable/admin/maintenance/backup_and_restore.html
https://docs.securedrop.org/en/stable/update_tails_usbs.html | https://docs.securedrop.org/en/stable/admin/maintenance/update_tails_usbs.html
https://docs.securedrop.org/en/stable/kernel_troubleshooting.html | https://docs.securedrop.org/en/stable/admin/maintenance/kernel_troubleshooting.html
https://docs.securedrop.org/en/stable/rebuild_admin.html | https://docs.securedrop.org/en/stable/admin/maintenance/rebuild_admin.html
https://docs.securedrop.org/en/stable/update_bios.html | https://docs.securedrop.org/en/stable/admin/maintenance/update_bios.html
https://docs.securedrop.org/en/stable/decommission.html | https://docs.securedrop.org/en/stable/admin/maintenance/decommission.html
https://docs.securedrop.org/en/stable/source.html | https://docs.securedrop.org/en/stable/source/source.html
https://docs.securedrop.org/en/stable/journalist.html | https://docs.securedrop.org/en/stable/journalist/journalist.html
https://docs.securedrop.org/en/stable/admin.html | https://docs.securedrop.org/en/stable/admin/reference/admin.html

I'm going to go through everything again to make sure I didn't miss any, but that should get us pretty close. One item worth noting is that because the original Admin guide has been split up into individual chapters, we may need to adjust the link in places like securedrop.org where one of the newer chapters would be more relevant. I'll try to proactively put that list together as well.

And sounds great about holding off for another approval. I'm going to ping @zenmonkeykstop just to make sure this is on his radar (although, of course, it's completely up to him if he wants to be the one to do the additional review, or leave it for another maintainer).

**Comment by rocodes** [comment]
Hey @nathandyer, took a spin through and it's looking great in terms of structure, I think it's a lot more readable and I really appreciate you for putting this together.

(I'm excited to chat with you about content/wording changes, as well as slimming down the docs, but that's a whole other can of worms. :) )

Small thing - found a 404 in the glossary: `https://docs.securedrop.org/en/latest/overview.html#infrastructure`. Otherwise LGTM :)

**Comment by nathandyer** [comment]
Thank you so much @rocodes! I'm looking forward to those conversations as well!

I appreciate you spotting that issue in the glossary. That should be fixed now.

Since we have two approvals I think we are OK to merge, but I will hold off on doing so until Monday (in part to give folks more time if they want to explore these changes first, and also because deploying major changes on Friday inevitably backfires :smiley: )

**Comment by nathandyer** [comment]
@eloquence @rocodes It looks like GitHub needs an official approval again since a new commit was pushed last week. Would either of you mind re-approving? Thank you!

**Review by eloquence — APPROVED** [review]
Re-approving and merging :). Let's make sure we have the redirects in place when we promote to `stable` - I'll open an infra issue for that now.

**Comment by cfm** [comment]
@nathandyer, this was on my review list for today, after your invitation last week, but Erik's approved and merged already.  :-)  So I'll just say that this reorganization now published to `latest` is a thing of beauty.  Thank you for advocating for it and working on it!

**Pull request merged**
